### PR TITLE
Update package dependencies

### LINKED ISSUE #1985: DEPS: Keycloak python client uses deprecated urllib3 module
A few developers, including myself, have run into this issue when trying to run our tests locally: https://github.com/urllib3/urllib3/issues/2680

Because we convert all warnings to errors, this was causing our tests to fail. The solution looks like upgrading the Keycloak python client to version 3.3.0.

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs

Closes #1985
I also pinned `setuptools` as a dev dependency to get around #1984   

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

**Comment by iameskild** [comment]
Given that we are also looking at upgrading Pydantic, this is blocked by #1928.

**Review by fangchenli — APPROVED** [review]

**Comment by fangchenli** [comment]
> Given that we are also looking at upgrading Pydantic, this is blocked by #1928.

We don't have to wait for #1928. For Pydantic, we could bump it to the last v1 version 1.10.12 for now and migrate to v2 after the next release.

**Review by fangchenli — APPROVED** [review]

**Comment by iameskild** [comment]
Thanks for reviewing @fangchenli! Unfortunately merging is still blocked. @pavithraes do you mind reviewing when you get a chance? Thank you both :)

**Review by sblair-metrostar — APPROVED** [review]
lgtm

**Review by pavithraes — APPROVED** [review]

**Comment by pavithraes** [comment]
Can merge after CI passes :)

File: pyproject.toml [hunk 1]
```diff
@@ -93,6 +94,7 @@ dev = [
     "grayskull",
     "build",
     "jinja2",
+    "setuptools==63.4.3",
```
Comment (aktech) [inline pyproject.toml:97]: Is there conflict/problem to upgrade this to latest (`68.2.2`)?

Comment (iameskild) [reply pyproject.toml:97]: This seems to be an issue with Keycloak: https://github.com/marcospereirampj/python-keycloak/pull/483

Being tracked here: https://github.com/nebari-dev/nebari/issues/1984

**Review by aktech — APPROVED** [review]

**Pull request merged**